BraveWallet/Crypto/Transaction Confirmations/TransactionConfirmationView.swift

```diff
@@ -21,21 +24,92 @@ struct TransactionConfirmationView: View {
   }
   
   @State private var viewMode: ViewMode = .transaction
-  @State private var activeTransactionIndex: Int = 0
+  @State private var activeTransactionId: BraveWallet.TransactionInfo.ID = "" {
+    didSet {
+      confirmationStore.fetchDetails(for: activeTransaction)
+    }
+  }
   
   private func next() {
-    activeTransactionIndex += 1
-    if activeTransactionIndex == transactions.endIndex {
-      activeTransactionIndex = 0
+    if let index = transactions.firstIndex(where: { $0.id == activeTransactionId }) {
+      var nextIndex = transactions.index(after: index)
+      if nextIndex == transactions.endIndex {
+        nextIndex = 0
+      }
+      activeTransactionId = transactions[nextIndex].id
+    } else {
+      activeTransactionId = transactions.first!.id
     }
   }
   
   private func rejectAll() {
-    presentationMode.dismiss()
+    for transaction in transactions {
+      confirmationStore.reject(transaction: transaction)
+    }
   }
   
   private var activeTransaction: BraveWallet.TransactionInfo {
-    transactions[activeTransactionIndex]
+    transactions.first(where: { $0.id == activeTransactionId }) ?? transactions.first!
+  }
+  
+  private var fromAccountName: String {
+    NamedAddresses.name(for: activeTransaction.fromAddress, accounts: keyringStore.keyring.accountInfos)
+  }
+  
+  private var toAccountName: String {
+    NamedAddresses.name(for: activeTransaction.txData.baseData.to, accounts: keyringStore.keyring.accountInfos)
+  }
+  
+  private var transactionType: String {
+    switch activeTransaction.txType {
+    case .erc20Approve:
+      return Strings.Wallet.transactionTypeApprove
+    default:
+      if activeTransaction.isSwap {
+        return Strings.Wallet.swap
+      }
+      return Strings.Wallet.send
+    }
+  }
+  
+  private var transactionDetails: String {
+    let data = activeTransaction.txData.baseData.data
+      .map { byte in
+        String(format: "%02X", byte.uint8Value)
```

Comment:
Would make sense to make a helper function 'DataExtensions' for that I think, is data to `0x` string conversion used anywhere else in the app?